contracts/FunctionsConsumer.sol

```diff
@@ -1,86 +1,80 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.7;
+pragma solidity ^0.8.19;
 
-import {Functions, FunctionsClient} from "./dev/functions/FunctionsClient.sol";
-// import "@chainlink/contracts/src/v0.8/dev/functions/FunctionsClient.sol"; // Once published
-import {ConfirmedOwner} from "@chainlink/contracts/src/v0.8/ConfirmedOwner.sol";
+import {FunctionsClient} from "./@chainlink/contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol";
+import {ConfirmedOwner} from "./@chainlink/contracts/src/v0.8/shared/access/ConfirmedOwner.sol";
+import {FunctionsRequest} from "./@chainlink/contracts/src/v0.8/functions/dev/1_0_0/libraries/FunctionsRequest.sol";
 
 /**
- * @title Functions Consumer contract
- * @notice This contract is a demonstration of using Functions.
- * @notice NOT FOR PRODUCTION USE
+ * @title Chainlink Functions example on-demand consumer contract example
  */
 contract FunctionsConsumer is FunctionsClient, ConfirmedOwner {
-  using Functions for Functions.Request;
+  using FunctionsRequest for FunctionsRequest.Request;
 
-  bytes32 public latestRequestId;
-  bytes public latestResponse;
-  bytes public latestError;
+  bytes32 public donId; // DON ID for the Functions DON to which the requests are sent
 
-  event OCRResponse(bytes32 indexed requestId, bytes result, bytes err);
+  bytes32 public s_lastRequestId;
+  bytes public s_lastResponse;
+  bytes public s_lastError;
+
+  error UnexpectedRequestID(bytes32 requestId);
+
+  constructor(address router, bytes32 _donId) FunctionsClient(router) ConfirmedOwner(msg.sender) {
+    donId = _donId;
+  }
 
   /**
-   * @notice Executes once when a contract is created to initialize state variables
-   *
-   * @param oracle - The FunctionsOracle contract
+   * @notice Set the DON ID
+   * @param newDonId New DON ID
    */
-  // https://github.com/protofire/solhint/issues/242
-  // solhint-disable-next-line no-empty-blocks
-  constructor(address oracle) FunctionsClient(oracle) ConfirmedOwner(msg.sender) {}
+  function setDonId(bytes32 newDonId) external onlyOwner {
+    donId = newDonId;
+  }
 
   /**
-   * @notice Send a simple request
-   *
+   * @notice Triggers an on-demand Functions request using remote encrypted secrets
    * @param source JavaScript source code
-   * @param secrets Encrypted secrets payload
-   * @param args List of arguments accessible from within the source code
-   * @param subscriptionId Funtions billing subscription ID
-   * @param gasLimit Maximum amount of gas used to call the client contract's `handleOracleFulfillment` function
-   * @return Functions request ID
+   * @param secretsLocation Location of secrets (only Location.Remote & Location.DONHosted are supported)
+   * @param encryptedSecretsReference Reference pointing to encrypted secrets
+   * @param args String arguments that will be passed into the source code
+   * @param bytesArgs Bytes arguments that will be passed into the source code as hex strings
```

Comment:
@param args String arguments that will be passed into the source code
@param bytesArgs Bytes arguments that will be passed into the source code as hex strings

Couple of thoughts:
1) both are optional so recommend removing "will" and just say "arguments to be passed..."
2) we may need to also consider smoothing out the language to distinguish when they'd use bytesArgs instead of just regular args

Reply:
args & bytesArgs are always passed, but if none are provided it will just be an empty array.

Reply:
For #2 I think that is beyond the scope of a code comment.  We should rely on docs for this.